test/index-test.ts

```diff
@@ -13,17 +17,15 @@ import {
   tap,
   propagateEventTask
 } from '@most/core'
-import { newDefaultScheduler, delay, asap } from '@most/scheduler'
-import { hold } from '../src/index'
 
-const collect = (stream, scheduler) => {
-  const eventValues = []
+const collect = <A>(stream: Stream<A>, scheduler: Scheduler) => {
+  const eventValues: A[] = []
   const collectStream = tap(x => eventValues.push(x), stream)
   return runEffects(collectStream, scheduler)
     .then(() => eventValues)
 }
 
-const verifyHold = f => {
+const verifyHold = (f: (stream: Stream<number>, scheduler: Scheduler) => Promise<unknown>) => {
```

Comment:
`Promise<unknown>` is probably ok here as a return type.  If we wanted to be a bit more accurate, `f`'s return type can probably also be universally quantified.  Off the top of my head, it seems like a more sophisticated type for `verifyHold` would be:

```typescript
const verifyHold = <A> (f: (stream: Stream<number>, scheduler: Scheduler) => Promise<A>): Promise<[number, A]> => {
```

Thoughts on whether there's value in being that specific?

Reply:
Fixed